enable-kdump/cos-enable-kdump.yaml

```diff
@@ -0,0 +1,95 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Deploy this DaemonSet to enable kdump on the COS nodes with the
+# "cloud.google.com/gke-kdump-enabled=true" label.
+#
+# WARNING: Disabling hyper-threading might have severe performance impact on
+# your clusters and application. Please ensure that this is acceptable before
+# deploying this to your production clusters.
+#
+# WARNING: Enabling kdump requires node reboot. Therefore, in order to avoid
+# disrupting your workloads, it is recommended to create a new node pool with
+# with the "cloud.google.com/gke-kdump-enabled=true" label in your cluster,
+# deploy the DaemonSet to enable kdump in that node pool, and then migrate
+# your workloads to the new node pool.
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+    name: enable-kdump
+    namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      name: enable-kdump
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        name: enable-kdump
+    spec:
+      volumes:
+      - name: host
+        hostPath:
+          path: /
+      initContainers:
+      - name: enable-kdump
+        image: ubuntu
+        command:
+        - /bin/bash
+        - -c
+        - |
+          function verify_base_image {
+            mount --bind /rootfs/etc/os-release /etc/os-release
+            local id="$(grep "^ID=" /host/etc/os-release)"
+            if [[ "${id#*=}" != "cos" ]]; then
+              echo "This kdump feature switch is designed to run on Container-Optimized OS only"
+              exit 1
```

Comment:
I think we should exit with success, otherwise, the pod will be recreated again and again because the init container fails.

Reply:
Good point. Thanks!